### ISSUE TITLE
Problem in using horovod distributed learning

### ISSUE THREAD

**Issue by TomorrowIsAnOtherDay** [ppwwyyxx/tensorpack#711]
Thanks for your Nice job for using Uber horovod as distributed learning optimizer~

But I encountered a problem when using the new trainer.
I have some ops which need to be executed after each epoch, e.g. shuffling data(all data were loaded in memory before training)
But only in chief worker(rank 0) these ops were executed.

**Comment by ppwwyyxx** [comment]
http://tensorpack.readthedocs.io/en/latest/modules/callbacks.html#tensorpack.callbacks.Callback.chief_only

**Comment by TomorrowIsAnOtherDay** [comment]
Thanks for your rapid reply, I will have a try：）

**Comment by TomorrowIsAnOtherDay** [comment]
It worked as expected.
Should it be a parameter when constructing a new callback ?

    data_shuffle_cb = CallbackFactory(trigger=lambda self: [data_reader.shuffle_train_dataset()])
    data_shuffle_cb.set_chief_only(False)

    return TrainConfig(
            data=TFDatasetInput(training_dataset),
            model=Policy(),
            callbacks=[
                data_shuffle_cb,
                InferenceRunner(TFDatasetInput(test_dataset), ScalarStats('cost')),
                ],  
            steps_per_epoch=train_num // args.batch * 4,
            max_epoch=80)

**Comment by ppwwyyxx** [comment]
Adding a new argument which most people won't care for ALL callbacks doesn't sound like a good idea to me -- it requires extra coding effort for every new callback to be written.

**Comment by TomorrowIsAnOtherDay** [comment]
Ok~ 💯
I will make it work in this way：）
